app/Decsys/Services/WebhookService.cs

```diff
@@ -55,30 +56,28 @@ public async Task Trigger(PayloadModel payload)
     /// <returns>True if filter matches</returns>
     private static bool FilterCriteria(WebhookModel webhook, PayloadModel payload)
     {
-        foreach (var eventType in webhook.TriggerCriteria.SelectMany(filter => filter.EventTypes))
+        foreach (var triggerCondition in webhook.TriggerCriteria)
         {
-            // check 1: if they are the same event types
-            var result = eventType.GetType() == payload.EventType.GetType();
-
-            if(result)
-                // check 2: type specific validation criteria
-                // add further criteria for further types
-                result = eventType switch {
-                    PageNavigation nav => CheckIsValid(nav, payload.EventType as PageNavigation),
-                    _ => false
-                };
-
-            return result;
+            if (triggerCondition.EventTypes.PageNavigation.Any() && payload.EventData.PageNavigation != null)
```

Comment:
I'm not sure this condition works quite right, but it needs changing anyway as PayloadModel and WebhookModel will contain fundamentally different structures for trigger conditions...

I think it should probably look more like the old code with a switch, but I might be wrong due to how we've structured trigger conditions.

I'll try and come back to it later with some clarity.